lms/djangoapps/courseware/module_render.py

```diff
@@ -475,13 +476,15 @@ def publish(block, event_type, event):
                 only_if_higher=event.get('only_if_higher'),
             )
         else:
-            aside_context = {}
+            context = contexts.course_context_from_course_id(course_id)
+            context['user_id'] = block.runtime.user_id
+            context['asides'] = {}
             for aside in block.runtime.get_asides(block):
                 if hasattr(aside, 'get_event_context'):
                     aside_event_info = aside.get_event_context(event_type, event)
                     if aside_event_info is not None:
-                        aside_context[aside.scope_ids.block_type] = aside_event_info
-            with tracker.get_tracker().context('asides', {'asides': aside_context}):
+                        context['asides'][aside.scope_ids.block_type] = aside_event_info
+            with tracker.get_tracker().context(event_type, context):
```

Comment:
Yes, I believe that is right.   Does it work?  

Also, is there any chance that the user_id in the runtime is missing or otherwise wrong?  This is a larger change in that it affects *all* events that come through here.  I'm wondering if it would be prudent to not override the user_id if we don't have a good value.  E.g. 
    if block.runtime.user_id: 
       context['user_id'] = block.runtime.user_id